nvflare/fuel/sec/cve_checker.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import re
+import ssl
+
+NO_VERSION = -1
+
+log = logging.getLogger(__name__)
+
+
+def parse_openssl_version(version: str) -> (int, int, int):
+
+    pattern = r'[a-zA-Z ]*([\d.]+).*'
+    matches = re.match(pattern, version)
+    if not matches:
+        log.error(f"Invalid OpenSSL version: {version}")
+        return None, None, None
+
+    parts = matches.group(1).split(".")
+    major = int(parts[0]) if len(parts) > 0 else NO_VERSION
+    minor = int(parts[1]) if len(parts) > 1 else NO_VERSION
+    patch = int(parts[2]) if len(parts) > 2 else NO_VERSION
+
+    return major, minor, patch
+
+
+def check_openssl(version: str = ssl.OPENSSL_VERSION) -> bool:
+    """Check if vulnerabilities exist in this version. It only checks against
+    most recent CVEs.
+
+    Args:
+        version: OpenSSL version string
+
+    Returns: True if vulnerabilities are found
+    """
+    triple = parse_openssl_version(version)
+
+    # Check CVE https://www.openssl.org/news/secadv/20221101.txt
+
+    # OpenSSL versions 3.0.0 to 3.0.6 are vulnerable to this issue.
+    if triple[0] == 3:
+        return (3, 0, 0) <= triple <= (3, 0, 6)
+
+    # OpenSSL 1.1.1 and 1.0.2 are not affected by this issue.
+    if triple[0] == 1:
+        return triple != (1, 1, 1) and triple != (1, 0, 2)
+
+    return False
+
+
+def warn_openssl(version: str = ssl.OPENSSL_VERSION):
```

Comment:
Instead of parsing the ssl.OPENSSL_VERSION string, it's better to use the ssl.OPENSSL_VERSION_INFO, which is a openssl version tuple.

Reply:
It's hard to run tests with that tuple so I decided to use the string. Also, the tuple is not well-documented, don't really know what the 5 numbers mean.